lib/mixins/acroform.js

```diff
@@ -0,0 +1,362 @@
+        } else if (f.type === 'percent') {
+          let p = Object.assign({}, FORMAT_DEFAULT.percent, f);
+          params = String([String(p.nDec), p.sepComma ? '0' : '1'].join(','));
+        }
+      }
+      opts.AA = opts.AA ? opts.AA : {};
+      opts.AA.K = {
+        S: 'JavaScript',
+        JS: new String(`${fnKeystroke}(${params});`)
+      };
+      opts.AA.F = {
+        S: 'JavaScript',
+        JS: new String(`${fnFormat}(${params});`)
+      };
+    }
+    delete opts.format;
+    return opts;
+  },
+
+  _formatParamsAsString(params) {
+    if (Array.isArray(params)) {
+      return JSON.stringify(params)
+        .replace('[', '')
+        .replace(']', '');
+    }
+    return String(params);
+  },
+
+  _resolveColors(opts) {
+    let color = this._normalizeColor(opts.backgroundColor);
+    if (color) {
+      if (!opts.MK) {
+        opts.MK = {};
+      }
+      opts.MK.BG = color;
+    }
+    color = this._normalizeColor(opts.borderColor);
+    if (color) {
+      if (!opts.MK) {
+        opts.MK = {};
+      }
+      opts.MK.BC = color;
+    }
+    delete opts.backgroundColor;
+    delete opts.borderColor;
+    return opts;
+  },
+
+  _resolveFlags(options) {
+    let result = 0;
+    Object.keys(options).forEach(key => {
+      if (FIELD_FLAGS[key]) {
+        result |= FIELD_FLAGS[key];
+        delete options[key];
+      }
+    });
+    if (result !== 0) {
+      options.Ff = options.Ff ? options.Ff : 0;
+      options.Ff |= result;
+    }
+    return options;
+  },
+
+  _resolveJustify(options) {
+    let result = 0;
+    if (options.align !== undefined) {
+      if (typeof FIELD_JUSTIFY[options.align] === 'number') {
+        result = FIELD_JUSTIFY[options.align];
+      }
+      delete options.align;
+    }
+    if (result !== 0) {
+      options.Q = result; // default
+    }
+    return options;
+  },
+
+  _resolveFont(options) {
+    // add current font to document-level AcroForm dict if necessary
+    if (this._acroform.fonts[this._font.id] === null) {
+      this._acroform.fonts[this._font.id] = this._font.ref();
+    }
+
+    // add current font to field's resource dict (RD) if not the default acroform font
+    if (this._acroform.defaultFont !== this._font.name) {
+      options.DR = { Font: {} };
+      options.DR.Font[this._font.id] = this._font.ref();
+    }
+    return options;
+  },
+
+  _resolveStrings(options) {
+    let choices = [];
+    function appendChoices(a) {
+      if (Array.isArray(a)) {
+        for (let idx = 0; idx < a.length; idx++) {
+          if (typeof a[idx] === 'string') {
+            choices.push(new String(a[idx]));
+          } else {
+            choices.push(a[idx]);
+          }
+        }
+      }
+    }
+
+    appendChoices(options.Opt);
+    appendChoices(options.choices);
+    if (choices.length) {
+      options.Opt = choices;
+    }
+
+    ['V', 'DV'].forEach(key => {
+      if (typeof options[key] === 'string') {
+        options[key] = new String(options[key]);
```

Comment:
No need to typecast to String here

Reply:
I think pdfkit uses this to determine whether to create a CosName or a CosString?